fhir-persistence-schema/src/main/java/com/ibm/fhir/schema/control/JavaBatchSchemaGenerator.java

```diff
@@ -0,0 +1,450 @@
+    public static final String M_FILTER = "M_FILTER";
+    public static final String INTERNALSTATUS = "INTERNALSTATUS";
+    public static final String USERDATA = "USERDATA";
+    public static final String M_PROCESSSKIP = "M_PROCESSSKIP";
+    public static final String M_READ = "M_READ";
+    public static final String M_READSKIP = "M_READSKIP";
+    public static final String M_ROLLBACK = "M_ROLLBACK";
+    public static final String M_WRITE = "M_WRITE";
+    public static final String M_WRITESKIP = "M_WRITESKIP";
+    public static final String ISPARTITIONEDSTEP = "ISPARTITIONEDSTEP";
+    public static final String FK_TOPLVL_STEPEXECID = "FK_TOPLVL_STEPEXECID";
+
+    // The schema holding all the Java Batch data
+    private final String schemaName;
+
+    /**
+     * Configures the destination schema for JavaBatch
+     * @param schemaName
+     */
+    public JavaBatchSchemaGenerator(String schemaName) {
+        this.schemaName = schemaName;
+    }
+
+    /**
+     * Create the tables needed by the Liberty JBatch databaseStore
+     * @param model
+     */
+    public void buildJavaBatchSchema(PhysicalDataModel model) {
+        addJobInstanceTable(model);
+        addJobExecutionTable(model);
+        addStepThreadExecutionTable(model);
+        addStepThreadInstanceTable(model);
+        addRemotablePartitionTable(model);
+        addGroupAssociationTable(model);
+        addJobParameterTable(model);
+    }
+
+    /**
+     * Adds the job table with the following values:
+     <code>
+     CREATE TABLE FHIR_BATCH.JOBPARAMETER (
+         NAME VARCHAR(255), 
+         VALUE VARCHAR(255),
+         FK_JOBEXECID BIGINT)
+     CREATE INDEX FHIR_BATCH.JP_FKJOBEXECID_IX ON FHIR_BATCH.JOBPARAMETER (FK_JOBEXECID)
+     ALTER TABLE FHIR_BATCH.JOBPARAMETER ADD CONSTRAINT JBPRMETERFKJBXECID FOREIGN KEY (FK_JOBEXECID)
+         REFERENCES FHIR_BATCH.JOBEXECUTION (JOBEXECID)
+     </code>
+     * @param model
+     */
+    public void addJobParameterTable(PhysicalDataModel model) {
+        // The default for the value is 255, and chosen to update to 4096
+        Table parameter = Table.builder(schemaName, JOBPARAMETER_TABLE)
+                .addVarcharColumn(NAME, 255, NULL) // NAME VARCHAR(255)
+                .addVarcharColumn(VALUE, 4096, NULL) // VALUE VARCHAR(255) (updated to 4096)
+                .addBigIntColumn(FK_JOBEXECID, NOT_NULL) // FK_JOBEXECID BIGINT
+                .addForeignKeyConstraintAltTarget("JBPRMETERFKJBXECID", schemaName, JOBEXECUTION_TABLE, JOBEXECID, FK_JOBEXECID)
+                .addIndex("JP_FKJOBEXECID_IX", FK_JOBEXECID)
+                .addPrivilege(FhirSchemaConstants.FHIR_BATCH_GRANT_GROUP, Privilege.SELECT)
+                .addPrivilege(FhirSchemaConstants.FHIR_BATCH_GRANT_GROUP, Privilege.INSERT)
+                .addPrivilege(FhirSchemaConstants.FHIR_BATCH_GRANT_GROUP, Privilege.DELETE)
+                .addPrivilege(FhirSchemaConstants.FHIR_BATCH_GRANT_GROUP, Privilege.UPDATE)
```

Comment:
maybe create these up front like we do in FHIRSchemaGenerator.  then, replaced these four lines with:
```
.addPrivileges(List<GroupPrivilege>)
```

Reply:
done